frontend/src/components/common/form/ParentSelect.tsx

```diff
@@ -38,29 +38,28 @@ export const ParentSelect: React.FC<IParentSelect> = ({
   isFilter,
   label,
 }) => {
-  const { setFieldValue, values } = useFormikContext();
+  const { setFieldValue, values, resetForm, dirty } = useFormikContext<any>();
   const value = getIn(values, field);
   /** control the selection of the typeahead component */
   const [selected, setSelected] = useState<SelectOption[]>([]);
-  /** determine whether the initial value has been loaded into the component or not */
-  const [loaded, setLoaded] = useState(false);
   /** used to trigger onBlur so menu disappears on custom header click */
   const [clear, setClear] = useState(false);
   /** select appropriate agency to set the field value to when present */
   const agency = value ? options.find(x => x.value === value.toString()) : null;
 
   useEffect(() => {
```

Comment:
The problem that I have with this is that it ends up setting a formik field using a set* method. This happens the instant the form is loaded, and triggers formik's dirty flag.

This causes a few issues with form integrity, but the main issue is that it ends up overwriting the persisted formik form before the user has a chance to load a draft from storage.